app/assets/javascripts/oxalis/controller/scene_controller.js

```diff
@@ -130,11 +138,56 @@ class SceneController {
 
     const meshMaterial = new THREE.MeshNormalMaterial();
     const mesh = new THREE.Mesh(geometry, meshMaterial);
-    this.rootGroup.add(mesh);
+    this.scene.add(mesh);
     this.stlMeshes[id] = mesh;
     this.updateMeshPostion(id, position);
   }
 
+  addIsosurface(vertices, segmentationId): void {
+    let geometry = new THREE.BufferGeometry();
+    geometry.addAttribute("position", new THREE.BufferAttribute(vertices, 3));
+
+    // convert to normal (unbuffered) geometry to merge vertices
+    geometry = new THREE.Geometry().fromBufferGeometry(geometry);
+    geometry.mergeVertices();
```

Comment:
Yes, this is the reason. As long as isosurfaces are small, this should not matter much. If we do this on the server, we are probably not (much) faster since this will increase the request time + we would need a new interface with a separate vertex list and index list and that kind of exposes the structure of the buffer geometry to the protocol.

Reply:
So I'd suggest measuring / monitoring the performance and only deal with is once it becomes a problem.